image-builder/builder-services/src/main/java/io/hyscale/builder/cleanup/services/impl/DeleteAfterBuildBinary.java

```diff
@@ -15,56 +15,66 @@
  */
 package io.hyscale.builder.cleanup.services.impl;
 
+import java.util.Arrays;
+import java.util.List;
+
 import org.apache.commons.lang3.StringUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Conditional;
 import org.springframework.stereotype.Component;
 
-import io.hyscale.builder.cleanup.services.ImageCleanupProcessor;
+import io.hyscale.builder.cleanup.services.DeleteAfterBuild;
+import io.hyscale.builder.services.impl.DockerBinaryImpl;
+import io.hyscale.builder.services.spring.DockerBinaryCondition;
 import io.hyscale.commons.commands.CommandExecutor;
 import io.hyscale.commons.commands.provider.ImageCommandProvider;
 import io.hyscale.commons.exception.HyscaleException;
 import io.hyscale.servicespec.commons.model.service.ServiceSpec;
 import io.hyscale.servicespec.commons.util.ImageUtil;
 
 /**
- * This class removes the service images from the host machine
- * which are built by hyscale. Hyscale adds a label to the
- * image as imageowner = hyscale. This clean up happends on all
- * those images which are tagged with the label imageowner=hyscale
- * <p>
+ * Class provides a docker binary based implementation of {@link DeleteAfterBuild}
+ * 
  * docker rmi $(docker images <serviceimage> --filter label=imageowner=hyscale -q)
  */
 
 @Component
-public class DeleteAfterBuild implements ImageCleanupProcessor {
+@Conditional(DockerBinaryCondition.class)
+public class DeleteAfterBuildBinary extends DeleteAfterBuild {
```

Comment:
Instead of this there should be DockerClient interface to push , pull, tag, build,remove etc. Use these operations in ImageBuilder and in the Cleanup Processor also.

Reply:
Impl of DockerClient will be DockerBinary and DockerRestClient . They should operate on high level operations of just push, pull, build, tag etc.

Reply:
We have a LocalImageBuildPushServiceImpl which will replace the ImageBuilder, there I can use the DockerClient to perform build, push, pull, etc?
For Image clean up, the docker client will have the remove option, but to get the imageIds to remove, we need different implementation of cleanUp policy which again would be based on Binary and REST client, is this approach fine?